app/controllers/oral_history_access_requests_controller.rb

```diff
@@ -0,0 +1,49 @@
+# PUBLIC FACING
+# Staff-facing actions are in app/controllers/admin/oral_history_access_requests_controller.rb
+class OralHistoryAccessRequestsController < ApplicationController
+
+  before_action :set_work
+
+  # GET /works/4j03d097t/request_oral_history_access
+  def new
+    @oral_history_access_request = cls.new
+  end
+
+  # POST "/request_oral_history_access"
+  def create
+    @oral_history_access_request = cls.new(oral_history_access_request_params)
+    @oral_history_access_request.work = @work
+    if @oral_history_access_request.save
+      redirect_to work_path(@work.friendlier_id), notice: 'Your request has been logged.'
+    else
+     render :new
+    end
+  end
+
+private
+  def cls
+    Admin::OralHistoryAccessRequest
+  end
+
+  def oral_history_access_request_params
+    params.require(:admin_oral_history_access_request).permit(
+      :work_friendlier_id, :patron_name, :patron_email,
+      :patron_institution, :intended_use, :status, :notes)
+  end
+
+  def set_work
+    if params['admin_oral_history_access_request'].present?
+      @work_friendlier_id = params['admin_oral_history_access_request'].delete('work_friendlier_id')
+    else
+      @work_friendlier_id = params['work_friendlier_id']
+    end
+    @work = Work.find_by_friendlier_id(@work_friendlier_id)
```

Comment:
This condition is because the ID comes in two different ways depending on whether it's the #new or #create action? 

Instead of one `before_action` with a condition, let's just inline this into the two actions? For instance, the first line of new could just be:

    @work = Work.find_by_friendlier_id(params['work_friendlier_id'])

In fact, also, use the verison with a `!` to get an exception raised if no work could be found: `Work.find_by_friendlier_id!(params['work_friendlier_id']`

Then you'd have to re-add the work for raising if the work is not marked requestable. But the net change will be FEWER lines code, trying to "DRY" with a before_action actually resulted in more code and more complicated code! Simplify it, have each action do what it needs.

Reply:
Good-- I wrestled with this for a bit and this is a better approach.